### DISCUSSION TITLE
Select multiple commits at once in "Search & Compare" to see diff of all commits at once

### DISCUSSION THREAD

**Discussion by sebsauer90** [gitkraken/vscode-gitlens#3469] [Ideas]
Hello everyone :)

The IntelliJ git integration has this functionality to search and select multiple commits at once, to see the diff of all selected commits at once. This feature is helpful for example for code reviews in projects with trunk based development. It would be awesome to have this functionality also as part of gitlens.

**Comment by evonnn222** [comment]
Yvonne

**Comment by sebastian-sauer-adesso** [comment]
any feedback here? :)

**Reply by eamodio** [reply]
It is definitely something we are looking into (as we want it as well). No ETA currently though,

**Reply by sebastian-sauer-adesso** [reply]
then I still looking forward :)

**Comment by phubinhdang** [comment]
Hi @sebastian-sauer-adesso 
I also miss IntelliJ’s ability to select and view combined diffs for multiple commits. While GitLens doesn’t yet have an exact equivalent, I’ve found a workaround that provides the same result. 

1. Open the Commit Graph (under the GitLens tab in VS Code).
2. Select a start commit and an end commit (hold Ctrl/Cmd to multi-select).
3. Right-click and choose "Compare Selected Commits".
4. The combined changes are shown in "Search & Compare" section. 

Below example taken on https://github.com/open-webui/open-webui.
<img width="3731" height="1198" alt="image" src="https://github.com/user-attachments/assets/7c9ec401-4a48-48e7-b6c0-e348272ebcf3" />
